### DISCUSSION TITLE
Creating `ChartEntryModelProducer` then quickly calling `setEntries` frequently fails

### DISCUSSION THREAD

**Discussion by autoreleasefool** [patrykandpatrick/vico#664] [Ideas]
This was not previously an issue, and is happening as of 1.13.0 I believe. Definitely not in 1.12.0, or at least not common enough that I noticed, but after upgrading to 1.13.0 it became frequent.

I believe it's related to the animation overhaul mentioned in the release notes. The `mutex.tryLock()` line fails. The workaround is simple enough, either pass your entries directly to the constructer instead of `setEntries`, or use `setEntriesSuspending`.

But it could be something to callout in release notes for those upgrading that this doesn't work as reliably as it used to!

**Comment by patrickmichalik** [comment]
Hello! Thanks for the report. This is an oversight on our part—we’ve resolved the issue in Vico 2 and will also be releasing a Vico 1 update that fixes it. In the meantime, either of the workarounds mentioned can be used.

By the way, I’d suggest updating to Vico 1.14.0, which includes quite a few important bug fixes.

**Comment by patrickmichalik** [comment]
[Vico 1.15.0 Alpha 1](https://github.com/patrykandpatrick/vico/releases/tag/v1.15.0-alpha.1) addresses this. Cheers!
